frontend/e2e/$lang+/_public/apply/adult/_adult-flow.spec.ts

```diff
@@ -0,0 +1,270 @@
+    await test.step('Type of application page', async () => {
+      await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/type-application/);
+      await expect(page.getByRole('heading', { level: 1, name: 'Type of application' })).toBeVisible();
+      await page.getByRole('button', { name: 'Continue' }).click();
+
+      //check for empty fields
+      await hasError(page, 'Select who this application is for');
+
+      await page.getByRole('radio', { name: 'I am applying for myself', exact: true }).check();
+      await page.getByRole('button', { name: 'Continue' }).click();
+    });
+
+    await test.step('Tax filing page', async () => {
+      await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/adult\/tax-filing/);
+      await expect(page.getByRole('heading', { level: 1, name: 'Tax filing' })).toBeVisible();
+      await page.getByRole('button', { name: 'Continue' }).click();
+
+      //check for empty fields
+      await hasError(page, 'Select whether or not you have filed your taxes');
+
+      await page.getByRole('radio', { name: 'No', exact: true }).check();
+      await page.getByRole('button', { name: 'Continue' }).click();
+    });
+
+    await test.step('File taxes page', async () => {
+      await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/adult\/file-taxes/);
+      await expect(page.getByRole('heading', { level: 1, name: 'File your taxes' })).toBeVisible();
+      await page.getByRole('link', { name: 'Back' }).click();
+      await page.getByRole('radio', { name: 'Yes', exact: true }).check();
+      await page.getByRole('button', { name: 'Continue' }).click();
+    });
+
+    await test.step('Date of birth page', async () => {
+      await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/adult\/date-of-birth/);
+      await expect(page.getByRole('heading', { level: 1, name: 'Date of birth' })).toBeVisible();
+      await page.getByRole('button', { name: 'Continue' }).click();
+
+      //check for empty fields
+      await hasError(page, 'Date of birth must include a year');
+      await hasError(page, 'Date of birth must include a month');
+      await hasError(page, 'Date of birth must include a day');
+
+      await fillOutDOB(page, '1990');
+      await page.getByRole('button', { name: 'Continue' }).click();
+    });
+
+    await test.step('Disability tax credit page', async () => {
+      await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/adult\/disability-tax-credit/);
+      await expect(page.getByRole('heading', { level: 1, name: 'Disability tax credit' })).toBeVisible();
+      await page.getByRole('button', { name: 'Continue' }).click();
+
+      //check for empty fields
+      await hasError(page, 'Select whether or not you have a valid disability tax credit');
+
+      await page.getByRole('radio', { name: 'No' }).check();
+      await page.getByRole('button', { name: 'Continue' }).click();
+    });
+
+    await test.step('Find out when you can apply page', async () => {
```

Comment:
This step should be split in 2 steps since clicking `Back` link returns the user to `Disability tax credit` page then you check `Yes` radio and click `Continue` on that page.